lte/gateway/c/core/oai/test/s1ap_task/test_s1ap_mme_handlers_with_injected_state.cpp

```diff
@@ -0,0 +1,181 @@
+    mme_app_handler = std::make_shared<MockMmeAppHandler>();
+    sctp_handler = std::make_shared<MockSctpHandler>();
+
+    itti_init(TASK_MAX, THREAD_MAX, MESSAGES_ID_MAX, tasks_info, messages_info,
+              NULL, NULL);
+
+    // initialize mme config
+    mme_config_init(&mme_config);
+    create_partial_lists(&mme_config);
+    mme_config.use_stateless = false;
+    hss_associated = true;
+
+    task_id_t task_id_list[4] = {TASK_MME_APP, TASK_S1AP, TASK_SCTP,
+                                 TASK_SERVICE303};
+    init_task_context(TASK_MAIN, task_id_list, 4, handle_message,
+                      &task_zmq_ctx_main_s1ap_with_injected_states);
+
+    std::thread task_mme_app(start_mock_mme_app_task, mme_app_handler);
+    std::thread task_sctp(start_mock_sctp_task, sctp_handler);
+    task_mme_app.detach();
+    task_sctp.detach();
+
+    s1ap_mme_init(&mme_config);
+
+    // add injection of state loaded in S1AP state manager
+    std::string magma_root = std::getenv("MAGMA_ROOT");
+    std::string state_data_path =
+        magma_root + "/" + DEFAULT_S1AP_STATE_DATA_PATH;
+    std::string data_folder_path =
+        magma_root + "/" + DEFAULT_S1AP_CONTEXT_DATA_PATH;
+    std::string data_list_path =
+        magma_root + "/" + DEFAULT_S1AP_CONTEXT_DATA_PATH + "data_list.txt";
+    assoc_id = 37;
+    stream_id = 1;
+    number_attached_ue = 2;
+
+    mock_read_s1ap_state_db(state_data_path);
+    name_of_ue_samples =
+        load_file_into_vector_of_line_content(data_folder_path, data_list_path);
+    mock_read_s1ap_ue_state_db(name_of_ue_samples);
+
+    state = S1apStateManager::getInstance().get_state(false);
+  }
+
+  virtual void TearDown() {
+    // Sleep to ensure that messages are received and contexts are released
+    std::this_thread::sleep_for(std::chrono::milliseconds(2000));
+
+    send_terminate_message_fatal(&task_zmq_ctx_main_s1ap_with_injected_states);
+    destroy_task_context(&task_zmq_ctx_main_s1ap_with_injected_states);
+    itti_free_desc_threads();
+
+    free_mme_config(&mme_config);
+
+    // Sleep to ensure that messages are received and contexts are released
+    std::this_thread::sleep_for(std::chrono::milliseconds(1500));
+  }
+
+ protected:
+  std::shared_ptr<MockMmeAppHandler> mme_app_handler;
+  std::shared_ptr<MockSctpHandler> sctp_handler;
+  s1ap_state_t* state;
+  sctp_assoc_id_t assoc_id;
+  sctp_stream_id_t stream_id;
+  std::vector<std::string> name_of_ue_samples;
+  int number_attached_ue;
+};
+
+TEST_F(S1apMmeHandlersWithInjectedStatesTest, GenerateUEContextReleaseCommand) {
+  ue_description_t ue_ref_p = {
+      .enb_ue_s1ap_id = 1,
+      .mme_ue_s1ap_id = 99,
+      .sctp_assoc_id = assoc_id,
+      .comp_s1ap_id = S1AP_GENERATE_COMP_S1AP_ID(assoc_id, 1)};
+  ue_ref_p.s1ap_ue_context_rel_timer.id = -1;
+  ue_ref_p.s1ap_ue_context_rel_timer.msec = 1000;
+
+  S1ap_S1AP_PDU_t pdu_s1;
+  memset(&pdu_s1, 0, sizeof(pdu_s1));
+  ASSERT_EQ(RETURNok, generate_s1_setup_request_pdu(&pdu_s1));
```

Comment:
Hi @thhung, May I know the significance of decoding s1_Setup request message. Because s1setup request is not handled.

Reply:
It is not required. However, from what I remember, for the version of Magma at the time I developed, without this, the resource of the test wasn't released properly. You can try to remove it. Maybe with the current version of Magma, we don't need this anymore.

Reply:
Sure! thanks for your response. I will check and update